inference-engine/thirdparty/clDNN/src/gpu/loop_gpu.cpp

```diff
@@ -0,0 +1,478 @@
+        }
+
+
+        // shortcut of execution_condition memory in body network
+        memory_impl* execution_condition_mem = nullptr;
+        if (node.is_execution_condition_used()) {
+            const primitive_id& condition_id = node.get_condition_id();
+            execution_condition_mem = &body_network->get_primitive(condition_id)->output_memory();
+        }
+
+        // output memory must be set before input_memory to set backedge memory properly
+        if (!instance.memroy_set) {
+            preprocess_output_memory(instance);
+            preprocess_input_memory(instance);
+            instance.memroy_set = true;
+        }
+
+        int64_t current_iteration = 0;
+        if (node.is_current_iteration_used()) {
+            write_int(*current_iteration_mem, current_iteration);
+        }
+        const auto& input_iteration_mem = instance.input_iteration_mem;
+        const auto& output_iteration_mem = instance.output_iteration_mem;
+        int actual_iteration = 0;
+        while (current_iteration < trip_count && execution_condition) {
+            // Copy & Set sliced input memory offset
+            for (size_t i = 0; i < instance.input_iteration_mem.size(); ++i) {
+                const auto& cropped_mem_info = input_iteration_mem.at(i);
+                memory_impl::ptr mem = cropped_mem_info.copy_and_get_sliced_mem(actual_iteration);
+                // set input mem
+                if (actual_iteration == 0) {
+                    body_network->set_input_data(cropped_mem_info.to_id, *mem);
+                } else {
+                    cropped_mem_info.to_prim->set_output_memory(*mem);
+                }
+            }
+
+            // Set backedged input
+            if (actual_iteration == 0) {
+                for (auto& edge_mem_bind : instance.backedge_mem) {
+                    const primitive_id& input_id = edge_mem_bind.to_primitive->id();
+                    body_network->set_input_data(input_id, *edge_mem_bind.initial_mem);
+                }
+            } else {
+                for (auto& edge_mem_bind : instance.backedge_mem) {
+                    edge_mem_bind.set_backedged_input(actual_iteration);
+                }
+            }
+
+            // Set sliced output memory offset
+            for (size_t i = 0; i < output_iteration_mem.size(); ++i) {
+                const auto& cropped_mem_info = output_iteration_mem.at(i);
+                const auto& from_mem = cropped_mem_info.sliced_mem.at(actual_iteration);
+                cropped_mem_info.from_prim->set_output_memory(*from_mem);
+            }
+            if (actual_iteration == 0) {
+                body_network->execute(events);
+            } else {
+                body_network->execute({});
+            }
+
+
+            // update index & execution condition for the next iteration
+            if (node.is_current_iteration_used()) {
+                current_iteration = read_int(*current_iteration_mem);
```

Comment:
BTW why should we use current_iteration as memory and read / write that at every iteration? Isnt it just be enough with variable? If it is to be used as "actual_iteration_count"  from outer network, I think it is enough to "write" there only when we stop the execution.

Reply:
Hmm I got something, "is_current_iteration_used" is trying to express that the # of iteration count might be variable length right? 
If so.. it is somewhat confusing. 
Lets rename the "is_current_iteration_used" to "is_fixed_seq_len" and 
if (is_fixed_seq_len) {
    current_iteration++; 
} else {
    write_int(*actual_iteration_mem, current_iteration);
}

Reply:
When I design the loop primitive api at first, I tried to support all the parameters defined in [IR loop operation spec](https://docs.openvinotoolkit.org/latest/openvino_docs_ops_infrastructure_Loop_5.html). `current_iteration`, `execution_condition` are the parameters defined in the API spec. `actual_iteration` is added to inform user the actual output size of the loop primitive (then user might crop the output size by actual_iteration). For the TI, current_itearation is equals to actual_iteration because body network never update current_iteration value and execution_condition. However, if we need to support generic loop, I believe we need to keep current_iteration.

Reply:
Also this is why reading and writing current_itearation and execution_condition is required.